zebra/zebra_dplane.c

```diff
@@ -2349,6 +2358,40 @@ dplane_route_update_internal(struct route_node *rn,
 #endif	/* !HAVE_NETLINK */
 		}
 
+		/*
+		 * If the old and new context type, and nexthop group id
+		 * are the same there is no need to send down a route replace
+		 * as that we know we have sent a nexthop group replace
+		 * or an upper level protocol has sent us the exact
+		 * same route again.
+		 */
```

Comment:
this is only being short circuited if the route is using the new API. That's because with this API, the upper proto should be sending a NHG replace rather than a route replace.

tagging @donaldsharp as well, he wrote this code